packages/patternfly-4/react-charts/src/components/ChartDonutThreshold/ChartDonutThreshold.js

```diff
@@ -0,0 +1,237 @@
+/* eslint-disable react/require-default-props,react/prop-types */
```

Comment:
The main reason is for composition and simplicity. 

I thought about combining the utilization and threshold charts, but the component was looking overly complex. Too many properties to support both the static and dynamic charts. And extra logic to modify the appearance of the dynamic pie with and without the outer thresholds.

Having one component would introduce multiple props for individual data sets, multiple assessors, multiple heights, widths, etc. Not exposing the full set of props for each pie would make customization difficult. IMO, using separate components for the static and dynamic pies is cleaner, makes composition more flexible, and provides functionality for specific use cases.

Reply:
That makes sense. I do like that it allows a means for customizing the static threshold donut. Since we are thinking about customization, what if we made this a bit more generic like `ChartDonutGroup` which will wrap any number of donut charts and nest them in the order they are provided. The first donut would be innermost, and last would be outermost. That way we can allow customized 'nested donuts' beyond just the threshold use case. Then in our utilization donut, we provide a default static `thresholdDonut` which can be overridden by the user. Obviously, providing a bunch of donuts would cause the size of the chart to grow, but we allow enough customization that the user could adjust accordingly.

Reply:
I like your thinking...

I had also thought about creating one more, top level component to wrap both threshold and utilization. Then provide thresholdComponent and utilizationComponent props to allow folks to override. Maybe some props for thresholdData, utilizationData, but then it started snowballing with accessors, height, width, etc.

I don't know if the above could be generic because the dynamic pie must be made smaller than normal in order to fit inside the threshold properly. The x and y coordinates are also slightly adjusted, etc. At least that particular layout is specific to the threshold chart.